### ISSUE TITLE
Shell expansion not available for password.fetch command

### ISSUE THREAD

**Issue by stweller** [pimutils/vdirsyncer#625]
I currently use a password.fetch command to get the password for a dav server. The config line I use looks like this:

`password.fetch = ["command", "gpg", "-d", "--use-agent", "/absolute/path/to/some/file]`

Now I would like to replace the absolute path by something like `~/relative/path/to/some/file`. This does now work, however, since the `~` gets fed literally to gpg, which doesn't expand it. If you ran this command through a shell, expansion of `~` (or `$HOME` or whatever) would be available.

vdirsyncer version used is 0.14.1 running with python 3.4.2 on Debian Linux.

Before you submit bug reports: https://vdirsyncer.pimutils.org/en/stable/contributing.html

**Comment by hobarrera** [comment]
This actually works for me:

     password.fetch = ["command", "~/.local/bin/vdirsyncer-pass", "fastmail:carddav"]

But you could actually run a shell that does the expanding:

     password.fetch = ["command", "bash", "-c", "$HOME/.local/bin/vdirsyncer-pass fastmail:carddav"]

**Comment by untitaker** [comment]
I can see how this is annoying, however this is exactly how I intended to implement it. There are two paths to choose when implementing this:

* Execute the command in the user's shell. That means that you'd write `["command", "gpg -d --use-agent foo"]`, which gives you the benefits like path expansion (like in your example), ability to reference environment variables, etc. However, it also means that you will have to take care of escaping/quoting spaces like you do in the shell, which can be annoying.
* Don't execute the command in a shell

I don't think behaving *partially* like a shell, i.e. expanding `~` but still treating spaces literally, is a good idea here, so I chose the latter option because it seemed like the stricter (and simpler) one.

What I usually recommend in this situation is to write `["command", "sh", "-c", "gpg -d --use-agent foo"]`. I think this should be documented better.
